x-pack/solutions/observability/plugins/inventory/public/hooks/use_discover_redirect.ts

```diff
@@ -37,19 +24,19 @@ export const useDiscoverRedirect = (entity: InventoryEntity) => {
         })
       : '';
 
-    return application.capabilities.discover?.show
+    return application.capabilities.discover?.show || !dataView
```

Comment:
@jennypavlova Sorry to show up on this PR after it's already merged, but I just encountered this line when resolving a merge conflict and thought it was worth verifying. If I'm reading this right, a Discover link will display if the user has the `show` capability or if there is no `dataView`. But couldn't the `|| !dataView` here cause the link to show in some cases even when the user doesn't have the required Discover capability?

Reply:
Thank you for the comment, good catch! I think we can remove the `|| !dataView` or make it an `AND` condition - this can be addressed as part of #207064 as it will be touching this part.